GherkinSyncTool/GherkinSyncTool/Synchronizers/SectionSynchronizer.cs

```diff
@@ -0,0 +1,115 @@
+    {
+        private static readonly Logger Log = LogManager.GetLogger(MethodBase.GetCurrentMethod()?.DeclaringType?.Name);
+        private readonly TestRailClientWrapper _testRailClientWrapper;
+        
+        public SectionSynchronizer(TestRailClientWrapper testRailClientWrapper)
+        {
+            _testRailClientWrapper = testRailClientWrapper;
+        }
+        
+        /// <summary>
+        /// Builds a tree structure for TestRail sections
+        /// </summary>
+        /// <param name="projectId">TestRail project Id</param>
+        /// <param name="suiteId">TestRail suite Id</param>
+        /// <returns></returns>
+        /// <exception cref="ArgumentException"></exception>
+        public IEnumerable<TestRailSection> GetSectionsTree(ulong projectId, ulong? suiteId)
+        {
+            if (suiteId is null) 
+                throw new ArgumentException($"SuiteId must be specified. Check the TestRail project #{projectId}");
+            
+            var testRailSections = _testRailClientWrapper.GetSections(projectId)
+                .Select(s=>new TestRailSection(s))
+                .ToDictionary(k => k.Id);
+            
+            var testRailCases = _testRailClientWrapper
+                .GetCases(projectId, suiteId.Value)
+                .GroupBy(k=>k.SectionId)
+                .ToDictionary(k=>k.Key, k=>k.ToArray());
+
+            var result = new List<TestRailSection>();
+            foreach (var section in testRailSections.Values)
+            {
+                if (testRailCases.TryGetValue(section.Id, out Case[] value))
+                    testRailSections[section.Id].FeatureFiles.AddRange(value);
+                if (section.ParentId != null)
+                    testRailSections[section.ParentId].ChildSections.Add(section);
+                else result.Add(section);
+            }
+            return result;
+        }
+
+        /// <summary>
+        /// Gets or creates TestRail section Id for selected .feature file
+        /// </summary>
+        /// <param name="path">Path to .feature file</param>
+        /// <param name="suiteId">TestRail suite Id</param>
+        /// <param name="projectId">TestRail project Id</param>
+        /// <returns></returns>
+        /// <exception cref="ArgumentNullException"></exception>
+        public ulong GetOrCreateSectionId(string path, ulong suiteId, ulong projectId)
+        {
+            var targetSections = GetSectionsTree(projectId, suiteId);
+            //Path includes name of the feature file and files catalog - hence SkipLast(1) and Skip(1)
+            var sourceSections = new Queue<string>(path.Split('\\').SkipLast(1).Skip(1));
+            var sectionId = GetOrCreateSectionIdRecursively(targetSections, sourceSections, suiteId, projectId);
+            if (sectionId is null)
+                throw new ArgumentNullException(nameof(sectionId), 
+                    "Section id cannot be null. You cannot put feature files in root folder");
```

Comment:
Is it possible that `_testRailClientWrapper.CreateSection()` method returns null?